torch/_functorch/aot_autograd.py

```diff
@@ -1502,11 +1542,18 @@ def inner_fn(*args):
         outs = list(outs)
         assert len(meta.output_info) == len(outs)
 
-        mutated_inputs_to_return = [
-            x
-            for (i, x) in enumerate(args_maybe_cloned)
-            if meta.input_info[i].mutates_metadata or meta.input_info[i].mutates_data
-        ]
+        if meta.is_train and meta.keep_input_mutations:
```

Comment:
I think this branch is also unnecessary?

also - this is inside of `fn_prepped_for_autograd`, so we should be guaranteed to be in the training case here